src/core/Object3D.js

```diff
@@ -115,8 +216,95 @@ class Object3D extends EventDispatcher {
 
 		this.userData = {};
 
+		this.updateMatrixSideEffects();
+
+	}
+
+	set matrixAutoUpdate( value ) {
+
+		this.matrixData.matrixAutoUpdate = value;
+
+	}
+
+	get matrixAutoUpdate() {
+
+		return this.matrixData.matrixAutoUpdate;
+
+	}
+
+	set matrixWorldNeedsUpdate( value ) {
+
+		this.matrixData.matrixWorldNeedsUpdate = value;
+
+	}
+
+	get matrixWorldNeedsUpdate() {
+
+		return this.matrixData.matrixWorldNeedsUpdate;
+
+	}
+
+	set matrixWorldAutoUpdate( value ) {
+
+		this.matrixData.matrixWorldAutoUpdate = value;
+
+	}
+
+	get matrixWorldAutoUpdate() {
+
+		return this.matrixData.matrixWorldAutoUpdate;
+
+	}
+
+	updateMatrixSideEffects() {
+
+		if ( this.onBeforeMatrixUpdate !== Object3D.prototype.onBeforeMatrixUpdate ) {
+
+			this.matrixData.updateMatrixWorldBefore = this.onBeforeMatrixUpdate.bind( this );
+
+		}
+
+		if ( this.onAfterMatrixUpdate !== Object3D.prototype.onAfterMatrixUpdate ) {
+
+			this.matrixData.updateMatrixWorldAfter = this.onAfterMatrixUpdate.bind( this );
+
+		}
+
+		if ( this.updateMatrixWorld !== Object3D.prototype.updateMatrixWorld ) {
+
+			console.warn( 'Do not override updateMatrixWorld() in Object3D sub-classes.' );
+			console.warn( 'To extend behaviour on updateMatrixWorld(), use onBeforeMatrixUpdate() and onAfterMatrixUpdate().' );
+
+			// We don't know whether the additional processing should be performed before or after updateMatrixWorld()
+			// so we call it twice.
+			// Slight performance hit, but more than compensated by the gains from monomorphic iteration.
+
+			this.matrixData.updateMatrixWorldBefore = function ( force ) {
+
+				this.onBeforeMatrixUpdate();
```

Comment:
Please check here and on line 296 if the functions are defined.

Reply:
Why is this necessary?

`onBeforeMatrixUpdate()` and `onAfterMatrixUpdate()` are included in the Object3D prototype, so will always exist for any Object3D sub-class.

Reply:
Also did not see that - this also should be fine then.